core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -135,19 +135,17 @@ case object GarbageCollectionMetrics extends ExecutorMetricType with Logging {
 
   override private[spark] def getMetricValues(memoryManager: MemoryManager): Array[Long] = {
     val gcMetrics = new Array[Long](names.length) // minorCount, minorTime, majorCount, majorTime
-    if (SparkEnv.get.conf.get(config.EVENT_LOG_GC_METRICS)) {
       ManagementFactory.getGarbageCollectorMXBeans.asScala.foreach { mxBean =>
-        if (youngGenerationGarbageCollector.contains(mxBean.getName)) {
-          gcMetrics(0) = mxBean.getCollectionCount
-          gcMetrics(1) = mxBean.getCollectionTime
-        } else if (oldGenerationGarbageCollector.contains(mxBean.getName)) {
-          gcMetrics(2) = mxBean.getCollectionCount
-          gcMetrics(3) = mxBean.getCollectionTime
-        } else {
-          logDebug(s"${mxBean.getName} is an unsupported garbage collector." +
-            s"Add it to ${config.EVENT_LOG_GC_METRICS_YOUNG_GENERATION_GARBAGE_COLLECTORS} " +
-            s"or ${config.EVENT_LOG_GC_METRICS_OLD_GENERATION_GARBAGE_COLLECTORS} to enable.")
-        }
+      if (youngGenerationGarbageCollector.contains(mxBean.getName)) {
+        gcMetrics(0) = mxBean.getCollectionCount
```

Comment:
is it ever possible that multiple collectors are considered young generation (or old generation)?  If so, you'd want to use `+=` here.

I'm not sure if there is a configuration where that might happen, but I'm also not super familiar with all variants here.  @kiszk maybe you know?

Reply:
AFAIK, one collector per generation is supported only.

Reply:
Sorry for being late. I agree with one collector per generation.